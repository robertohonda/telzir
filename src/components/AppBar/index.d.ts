import React from 'react'
import { StandardProps } from '@material-ui/core';
import { AppBarProps } from '@material-ui/core/AppBar';

export interface CustomAppBarProps extends StandardProps<AppBarProps, AppBarClassKey> {
  logoPath: string
  title: string
}

export type AppBarClassKey =
  | 'toolbar'
  | 'title'
  | 'logo'

declare const CustomAppBar: React.ComponentType<CustomAppBarProps>

export default CustomAppBar