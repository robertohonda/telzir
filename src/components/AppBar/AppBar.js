import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import PropTypes from 'prop-types'

const useStyles = makeStyles((theme) => ({
  toolbar: {
    display: 'flex',
    justifyContent: 'center',
  },
  title: {
    marginLeft: theme.spacing(0.5),
    fontSize: 20,
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing(0.5),
      fontSize: 24,
    },
  },
  logo: {
    height: 40,
    [theme.breakpoints.up('sm')]: {
      height: 50,
    },
  },
}))

const CustomAppBar = (props) => {
  const classes = useStyles(props)
  const { logoPath, title, ...appBarProps } = props

  return (
    <AppBar {...appBarProps}>
      <Toolbar className={classes.toolbar}>
        <img className={classes.logo} alt="logo" src={logoPath} />
        <Typography className={classes.title} variant="h6">
          {title}
        </Typography>
      </Toolbar>
    </AppBar>
  )
}

CustomAppBar.propTypes = {
  logoPath: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
}

export default CustomAppBar
