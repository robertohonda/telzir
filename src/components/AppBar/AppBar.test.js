import React from 'react'
import { shallow } from 'enzyme'
import AppBar from './AppBar'

describe('AppBar', () => {
  it('should render correctly', () => {
    const component = shallow(<AppBar logoPath="logo.png" title="Telzir" />)

    expect(component).toMatchSnapshot()
  })
})
