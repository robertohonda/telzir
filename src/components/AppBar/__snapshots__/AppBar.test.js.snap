// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should render correctly 1`] = `
<WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Toolbar))
    className="makeStyles-toolbar-1"
  >
    <img
      alt="logo"
      className="makeStyles-logo-3"
      src="logo.png"
    />
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-title-2"
      variant="h6"
    >
      Telzir
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
